### PR TITLE
t_117: added method finishCrowdsaleBurnUnused to contract

### PR DESCRIPTION
**Pull request by avgoeid**

### CODE REVIEW

File: contracts/ICO_controller.sol [hunk 1]
```diff
@@ -215,24 +215,30 @@ contract ICO_controller is Ownable, TransferableInterface {
...
+        if (totalSold >= SOFTCAP) {
             // burn some unspent reward tokens
             token.burn(MAX_DEV_REWARD.sub(totalDevReward));
             // burn after airdrop left tokens
             if (totalAirdropAdrresses != 0) {
                uint256 airdropToBurn = AIRDROP_SUPPLY.sub(AIRDROP_SUPPLY.div(totalAirdropAdrresses).mul(totalAirdropAdrresses));
                 if (airdropToBurn != 0){
                     token.burn(airdropToBurn);
-                } 
+                }
```
Comment (serebrov) [inline contracts/ICO_controller.sol]: Please also check before `token.burn(MAX_DEV_REWARD.sub(totalDevReward))` that we are trying to burn > 0 tokens and that we have more tokens on the balance than we are trying to burn.

Also when burning airdrop leftovers - check that we have enough tokens on the balance.

File: test/test_stages.js [hunk 2]
```diff
@@ -414,3 +415,59 @@ contract('ICO success', async function (accounts) {
...
+        await controller_instance.finishCrowdsale();
+        let controllerTokenBalance = BigNumber(await token.balanceOf(controller_instance.address));
+        let marketingSupportTokens = BigNumber(await controller_instance.MARKETING_SUPPORT_SUPPLY.call());
+        let totalDevReward = await controller_instance.totalDevReward.call();
+        let maxDevReward = await controller_instance.MAX_DEV_REWARD.call();
+        let airDropSupply = await controller_instance.AIRDROP_SUPPLY.call();
+        assert.isTrue(
+            controllerTokenBalance.isEqualTo(marketingSupportTokens
+                .plus(maxDevReward.minus(totalDevReward).plus(airDropSupply))), 
+            "Wrong amount of token at controller(there are unburned tokens)"
```
Comment (serebrov) [inline test/test_stages.js]: I think the message here should be just "Wrong amount of tokens at controller".

**Review by serebrov — APPROVED** [review]
Looks good, please check my comments.

File: contracts/ICO_controller.sol [hunk 3]
```diff
@@ -215,24 +215,30 @@ contract ICO_controller is Ownable, TransferableInterface {
...
+            // send other 50% to multisig holder address
+            holder.transfer(this.balance);
+        }
+        crowdsaleFinished = true;
+
+    }
+
+    function finishCrowdsaleBurnUnused() external onlyOwner {
+        require(crowdsaleFinished);
+        if (totalSold >= SOFTCAP) {
```
Comment (Norfolks) [inline contracts/ICO_controller.sol]: No sense to check SOFTCAP here

**Pull request merged**